app/src/main/java/chat/rocket/android/service/RocketChatWebSocketThread.java

```diff
@@ -123,111 +137,149 @@ private void forceInvalidateTokens() {
     }).continueWith(new LogcatIfError());
   }
 
-  @Override
-  public boolean quit() {
+  /**
+   * terminate WebSocket thread.
+   */
+  @DebugLog
+  public Single<Boolean> terminate() {
     if (isAlive()) {
-      new Handler(getLooper()).post(() -> {
-        RCLog.d("thread %s: quit()", Thread.currentThread().getId());
-        unregisterListeners();
-        RocketChatWebSocketThread.super.quit();
+      return Single.fromEmitter(emitter -> {
+        new Handler(getLooper()).post(() -> {
+          RCLog.d("thread %s: terminated()", Thread.currentThread().getId());
+          unregisterListeners();
+          connectivityManager.notifyConnectionLost(hostname,
+              ConnectivityManagerInternal.REASON_CLOSED_BY_USER);
+          RocketChatWebSocketThread.super.quit();
+          emitter.onSuccess(true);
+        });
       });
-      return true;
     } else {
-      return super.quit();
+      connectivityManager.notifyConnectionLost(hostname,
+          ConnectivityManagerInternal.REASON_NETWORK_ERROR);
+      super.quit();
+      return Single.just(true);
     }
   }
 
   /**
+   * THIS METHOD THROWS EXCEPTION!!
+   * Use terminate() instead!!
+   */
+  @Deprecated public final boolean quit() {
```

Comment:
This method is really not intended to use.
So I want to add some kind of annotation for indicating it.